Assets/_CueClash/Scripts/Shooting/Weapon.cs

```diff
@@ -0,0 +1,24 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using TMPro;
+
+public class Weapon : MonoBehaviour
+{
+    public Transform bulletSpawnPoint;
+    public GameObject bulletPrefab;
+    public float bulletSpeed = 20;
+    public int nrOfBullets = 10;
+    public AmmoText ammoTextBox;
+
+    private void Update()
+    {
+        if (Input.GetKeyDown(KeyCode.Mouse0) && nrOfBullets > 0)

```

Comment:
Schimba sa folosesti InputHandler. Also baga faza cu inputul in playerobject. Baga restul intr-o functie. Ia in playerobject referinta la weapon si apeleaza de acolo functia de shoot.